sdks/python/apache_beam/io/bigquery.py

```diff
@@ -145,7 +146,14 @@ class RowAsDictJsonCoder(coders.Coder):
   """
 
   def encode(self, table_row):
-    return json.dumps(table_row)
+    # The normal error when dumping NAN/INF values is:
+    # ValueError: Out of range float values are not JSON compliant
+    # This code will catch this error to emit an error that explains
+    # to the programmer that they have used NAN/INF values.
+    try:
+      return json.dumps(table_row, allow_nan=False)
```

Comment:
We could try running bigquery_tornadoes example using DataflowPipelineRunner. If you don't see a significant performance difference there we should be good to go. I was just worried if this adds some significantly slow per-element check.